runner/app/live/pipelines/comfyui.py

```diff
@@ -52,52 +52,62 @@ def validate_prompt(cls, v) -> dict:
 class ComfyUI(Pipeline):
     def __init__(self):
         comfy_ui_workspace = os.getenv(COMFY_UI_WORKSPACE_ENV)
-        self.client = ComfyStreamClient(cwd=comfy_ui_workspace)
+        self.pipeline = ComfyStreamPipeline(width=512, height=512, cwd=comfy_ui_workspace)
         self.params: ComfyUIParams
-        self.video_incoming_frames: asyncio.Queue[VideoOutput] = asyncio.Queue()
 
     async def initialize(self, **params):
         new_params = ComfyUIParams(**params)
         logging.info(f"Initializing ComfyUI Pipeline with prompt: {new_params.prompt}")
-        # TODO: currently its a single prompt, but need to support multiple prompts
-        await self.client.set_prompts([new_params.prompt])
+        await self.pipeline.set_prompts([new_params.prompt])
         self.params = new_params
 
         # Warm up the pipeline
-        dummy_frame = VideoFrame(None, 0, 0)
-        dummy_frame.side_data.input = torch.randn(1, 512, 512, 3)
-
-        for _ in range(WARMUP_RUNS):
-            self.client.put_video_input(dummy_frame)
-            _ = await self.client.get_video_output()
+        await self.pipeline.warm_video()
         logging.info("Pipeline initialization and warmup complete")
 
+
     async def put_video_frame(self, frame: VideoFrame, request_id: str):
-        image_np = np.array(frame.image.convert("RGB")).astype(np.float32) / 255.0
-        frame.side_data.input = torch.tensor(image_np).unsqueeze(0)
-        frame.side_data.skipped = True
-        self.client.put_video_input(frame)
-        await self.video_incoming_frames.put(VideoOutput(frame, request_id))
-
-    async def get_processed_video_frame(self):
-        result_tensor = await self.client.get_video_output()
-        out = await self.video_incoming_frames.get()
-        while out.frame.side_data.skipped:
-            out = await self.video_incoming_frames.get()
```

Comment:
Don't we need this logic anymore?

Reply:
> Don't we need this logic anymore?

The frame skipping logic is in comfystream here https://github.com/livepeer/comfystream/blob/e9b87bc7bc0adb32c5731f70b1e11e44c1f2e3c1/src/comfystream/pipeline.py#L169-L170

I also fixed the `request_id` side_data passthrough